test/integration_test.go

```diff
@@ -4123,6 +4124,17 @@ func TestReadTerragruntConfigIamRole(t *testing.T) {
 	assert.True(t, util.FileNotExists(util.JoinPath(TEST_FIXTURE_READ_IAM_ROLE, identityArn+".txt")))
 }
 
+func TestReadTerragruntAuthProviderCmd(t *testing.T) {
+	t.Parallel()
+
+	cleanupTerraformFolder(t, TEST_FIXTURE_AUTH_PROVIDER_CMD)
+	tmpEnvPath := copyEnvironment(t, TEST_FIXTURE_AUTH_PROVIDER_CMD)
+	rootPath := util.JoinPath(tmpEnvPath, TEST_FIXTURE_AUTH_PROVIDER_CMD)
+	mockPiplinesCmd := filepath.Join(rootPath, "mock-pipelines.sh")
+
+	runTerragrunt(t, fmt.Sprintf("terragrunt run-all init --terragrunt-working-dir %s --terragrunt-auth-provider-cmd %s", rootPath, mockPiplinesCmd))
```

Comment:
If you're looking for a way to prove that this is working, you can just output the following from the script:

```json
{
  "envs": {
    "TF_VAR_foo": "bar"
  }
}
```

And have a `variable` `foo`, then read it in the outputs.

All we need to prove is that we can set some environment variables before the call to OpenTofu/Terraform is done.

Reply:
Considering that we have decided to use a strictly defined structure,  I have to use `{awsCredentials: {ACCESS_KEY_ID: ....` in the test